lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -838,11 +865,17 @@ sub fix_tests{
     my $table = $check->table;
     my $where = $check->id_where;
     my @edits = @{$check->columns};
+    # If we are inserting and id is displayed, we want to insert
+    # at this exact location
+    my $id_displayed = $check->{insert}
+                and grep( /^$check->{id_column}$/, @{$check->display_cols} );
 
     my $query;
-    if ($request->{insert}) {
-        my $columns = join(', ', map { $dbh->quote_identifier($_) } @edits);
-        my $values = join(', ', map { '?' } @edits);
+    if ($check->{insert}) {
+        my @id = $id_displayed ? $check->{id_column} : ();
```

Comment:
Investigating more: there's no case (currently) where `id` isn't displayed. What would be a suitable `insert` scenario where `id` isn't displayed?